### PR TITLE
Add SAI Notification support for host_tx_ready

### PR DESCRIPTION
**Pull request by noaOrMlnx**
New PORT_HOST_TX_READY notification logic was added to sonic-sairedis and syncd in order to support host_tx_ready synchronization enhancements. 

HLD: https://github.com/sonic-net/SONiC/pull/1453

### CODE REVIEW

File: meta/Meta.cpp [hunk 1]
```diff
@@ -6434,6 +6434,51 @@ void Meta::meta_sai_on_nat_event(
...
+                host_tx_ready_status);
+
+        return;
+    }
+
+    auto ot = objectTypeQuery(port_id);
+    auto valid = false;
+    valid = (ot == SAI_OBJECT_TYPE_PORT);
+
+    if (!valid)
```
Comment (kcudnik) [inline meta/Meta.cpp:6458]: used only once, you can put condition in te same line as if

File: syncd/NotificationProcessor.cpp [hunk 2]
```diff
@@ -463,6 +463,28 @@ void NotificationProcessor::process_on_queue_deadlock_event(
...
+void NotificationProcessor::process_on_port_host_tx_ready_change(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_object_id_t port_id,
+        _In_ sai_port_host_tx_ready_status_t *host_tx_ready_status)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_DEBUG("Port ID before translating from RID to VID is 0x%lx", port_id);
+    sai_object_id_t port_vid = m_translator->translateRidToVid(port_id, SAI_NULL_OBJECT_ID);
+    SWSS_LOG_DEBUG("Port ID after translating from RID to VID is 0x%lx", port_vid);
```
Comment (kcudnik) [inline syncd/NotificationProcessor.cpp:476]: you can use sai_serialize_object_id(port_id).c_str() to have string which will also add oid: prefix

File: syncd/scripts/syncd_init_common.sh [hunk 3]
```diff
@@ -233,6 +233,7 @@ config_syncd_mlnx()
     if [[ -f /tmp/sai_extra.profile ]]; then
         cat /tmp/sai_extra.profile >> /tmp/sai.profile
     fi
+
```
Comment (kcudnik) [inline syncd/scripts/syncd_init_common.sh:236]: this empty line is no t needed

File: meta/Meta.cpp (hunk 1 above)
Comment (noaOrMlnx) [reply meta/Meta.cpp:6458]: Done

File: syncd/NotificationProcessor.cpp (hunk 2 above)
Comment (noaOrMlnx) [reply syncd/NotificationProcessor.cpp:476]: Done

File: syncd/scripts/syncd_init_common.sh (hunk 3 above)
Comment (noaOrMlnx) [reply syncd/scripts/syncd_init_common.sh:236]: Done

**Review by Junchao-Mellanox — APPROVED** [review]

**Comment by kcudnik** [comment]
hey, you must update to latest master to fix notifications since SAI was updated, you can take a look here: https://github.com/sonic-net/sonic-sairedis/pull/1311/files

**Comment by kcudnik** [comment]
please fix whitespaces:
```
../vslib/Switch.cpp:                m_switchNotifications.on_port_host_tx_ready = 
ERROR: some files contain white spaces at the end of line, please fix
FAIL: checkwhitespace.sh
```

**Comment by noaOrMlnx** [comment]
Hi @kcudnik 

I tried to remove all whitespaces, and there is no whitespace in the line stated in the UT. 
Can you please point me to the issue?

**Comment by kcudnik** [comment]
Please spend time to read compiler ero
Rorschach in AZ pipeline:
```
Checkig for white spaces ...
../saiplayer/SaiPlayer.cpp:    
../saiplayer/SaiPlayer.h:        
../vslib/Switch.cpp:                m_switchNotifications.on_port_host_tx_ready = 
ERROR: some files contain white spaces at the end of line, please fix
FAIL: checkwhitespace.sh
```

**Review by prgeor — APPROVED** [review]

File: unittest/lib/TestSwitch.cpp [hunk 4]
```diff
@@ -36,6 +36,7 @@ TEST(Switch, updateNotifications)
     attrs[5].value.ptr = (void*)1;
     attrs[6].value.ptr = (void*)1;
     attrs[7].value.ptr = (void*)1;
+    attrs[8].value.ptr = (void*)1;
```
Comment (kcudnik) [inline unittest/lib/TestSwitch.cpp:39]: missing value for [9]

**Pull request merged**